code/modules/shuttle/misc/shuttle_remote.dm

```diff
@@ -0,0 +1,126 @@
+/obj/item/shuttle_remote
+	name = "shuttle remote"
+	desc = "A remote to send away or call a shuttle."
+	icon = 'icons/obj/devices/voice.dmi'
+	icon_state = "walkietalkie"
+	w_class = WEIGHT_CLASS_SMALL
+	///if the docks may be changed
+	var/may_change_docks = TRUE
+	///the port where the shuttle leaves to
+	var/shuttle_away_id = "whiteship_away"
+	///the port where the shuttle returns to
+	var/shuttle_home_id = "whiteship_home"
+	///var which will hold the nav computer
+	var/obj/machinery/computer/shuttle/our_computer
+	///var which will hold the mobile port
+	var/obj/docking_port/mobile/our_port
+
+/obj/item/shuttle_remote/examine(mob/user)
+	. = ..()
+	if(may_change_docks && our_computer)
+		. += span_notice("You can change where the [get_area_name(SSshuttle.getShuttle(our_computer.shuttleId))] docks using [EXAMINE_HINT("alt-right-click")].")
+
+/obj/item/shuttle_remote/Initialize(mapload)
+	. = ..()
+	if(!mapload)
+		return
+	var/obj/machinery/computer/shuttle/computer = locate(/obj/machinery/computer/shuttle) in loc
+	if(!computer)
+		log_mapping("[src] failed to find a navigation computer at [AREACOORD(src)]")
+		return
```

Comment:
Surely we should be able to map a remote down without having an associated computer. For like a shuttle workshop ruin or something.